### DISCUSSION TITLE
Add APPLY button to Preferences Menu

### DISCUSSION THREAD

**Discussion by Chorbu** [jankae/LibreVNA#203] [Ideas]
Hello Jan,

Is it possible to add APPLY button to the entire PREFERENCES Menus (as you made in Axis setup for my previous request) ? It will be more easy to test different options.

Sorry but i don't know if i write my request on the good discussion way ?

Best regards
Patrick

**Comment by jankae** [comment]
Hi Patrick,
the preferences window now has an apply button as well, thanks for the suggestion. Some settings (such as the SCPI server configuration) is not directly affected when pressing apply, it will only update after the preferences window has been closed. That is not on purpose but changing it would need some really ugly software work-arounds. Most other stuff (such as graph colors) is updated immediately.

You can post ideas here in the discussion section but if you want to make sure they stay on my radar until implemented, creating an issue over [here](https://github.com/jankae/LibreVNA/issues) is the better way (issues can be used for feature requests as well, not just for reporting problems)

**Reply by Chorbu** [reply]
Hi Jan,

Thank you for this button, it is great ! Ok for the way to post requests and reporting problems.

Just a questions about special marker formats :

Are there all operational ? Like P1dB (maybe dépend of the way to mesure 1dB compression point) ?
One S21 trace in PAUSE mode and the other in LIVE mode, increasing power entering the amplifier to reach 1dB loss between two traces ?


Thank you for modification you've done
Best regards
Patrick

**Comment by jankae** [comment]
> Are there all operational ?

Yes, all implemented marker formats should be fully working. For P1dB, you need to do a power sweep (with the frequency constant and the stimulus power changing). You can configure that in the sweep toolbar. Once you have a S21 (or S12) trace from the power sweep, you should be able to change a marker to P1dB type on that trace.

And yes, the idea is to find the power level at which the transmission drops by 1dB but instead of doing it over multiple (frequency) sweeps, it is done on a single power sweep.

**Comment by Chorbu** [comment]
Hi Jan,

Yes Ok for this explanation. I 'll try on this way opposite to multiple frequency in Power mode.

For my previous problems in filters with IF BW i remember i used to have no spikes before with 1KHz IF BW with these filters. So maybe characteristics of my VNA is degrading.

Do you plan to sell your LIBRECAL Soon ?

Thank you
Best regards
Patrick
